nixos/modules/services/misc/gotenberg.nix

```
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gotenberg;
+
+  args =
+    [ "--api-port=${toString cfg.port}" ]
+    ++ optionals (cfg.timeout != null) [ "--api-timeout=${cfg.timeout}" ]
+    ++ optionals (cfg.rootPath != null) [ "--api-root-path=${cfg.rootPath}" ]
+    ++ optionals (cfg.enableBasicAuth) [ "--api-enable-basic-auth" ]
+    ++ optionals (cfg.chromium.maxQueueSize != 0) [
+      "--chromium-max-queue-size=${toString cfg.chromium.maxQueueSize}"
+    ]
+    ++ optionals (cfg.chromium.autoStart) [ "--chromium-auto-start" ]
+    ++ optionals (cfg.chromium.disableJavascript) [ "--chromium-disable-javascript" ]
+    ++ optionals (cfg.chromium.disableRoutes) [ "--chromium-disable-routes" ]
+    ++ optionals (cfg.libreoffice.restartAfter != 10) [
+      "--libreoffice-restart-after=${toString cfg.libreoffice.restartAfter}"
+    ]
+    ++ optionals (cfg.libreoffice.maxQueueSize != 0) [
+      "--libreoffice-max-queue-size=${toString cfg.libreoffice.maxQueueSize}"
+    ]
+    ++ optionals (cfg.libreoffice.autoStart) [ "--libreoffice-auto-start" ]
+    ++ optionals (cfg.libreoffice.disableRoutes) [ "--libreoffice-disable-routes" ]
+    ++ optionals (cfg.pdfEngines != null) [
+      "--pdfengines-engines=${lib.concatStringsSep "," cfg.pdfEngines}"
+    ]
+    ++ optionals (cfg.logLevel != "info") [ "--log-level=${cfg.logLevel}" ];
```

eg

```suggestion
    ++ optional (cfg.logLevel != "info") "--log-level=${cfg.logLevel}";
```


but what speaks against setting those settings explicitly all the time like?:

```suggestion
    ++ [ "--log-level=${cfg.logLevel}" ];
```

and then ofcourse combine them into one list

---

> but what speaks against setting those settings explicitly all the time like?:
Honestly it's just a personal preference that if they're default then don't add them(I think it makes the command line look a bit nicer, but I guess there's no reason to not add them.